integration-tests/tests/src/tests/queries.ts

```diff
@@ -15,129 +15,1037 @@
-    primitives = realm.objects<IPrimitive>(PrimitiveSchema.name);
+type QueryLengthPair = [ExpectedLength: number, Query: string, ...QueryArgs: Array<any>];
+type QueryExceptionPair = [ExpectedException: string, Query: string, ...QueryArgs: Array<any>];
+type QueryResultsPair = [ExpectedResults: any[], Query: string, ...QueryArgs: Array<any>];
+
+describe("Queries", () => {
+  /**
+   * Helper method which runs an array of queries and asserts them to given expected length
+   * For example: (r, "Obj", [1, "intCol == 0"]) => querying "intCol == 0" should return 1 elements.
+   */
+  const assertQueryLength = (
+    realm: Realm,
+    objectSchema: string | Realm.ObjectClass,
+    queryLengthPairs: QueryLengthPair[],
+  ) => {
+    queryLengthPairs.forEach(([expectedLength, queryString, ...queryArgs]) => {
+      const filtered = realm.objects(objectSchema).filtered(queryString, ...queryArgs);
+      expect(filtered.length).equal(
+        expectedLength,
+        `Expected length ${expectedLength} for query: ${queryString} ${JSON.stringify(queryArgs)}`,
+      );
+    });
+  };
+
+  /**
+   * Helper method which runs an array of queries and asserts them to throw given exception.
+   * For example: (r, "Obj", ["invalid", "intCol == #"]) => querying "intCol == #" should
+   * throw an exception which includes the string "invalid".
+   */
+  const assertQueryException = (
+    realm: Realm,
+    objectSchema: string | Realm.ObjectClass,
+    queryExceptionPairs: QueryExceptionPair[],
+  ) => {
+    queryExceptionPairs.forEach(([expectedException, queryString, ...queryArgs]) => {
+      expect(() => {
+        realm.objects(objectSchema).filtered(queryString, ...queryArgs);
+      }).throws(
+        expectedException,
+        `Expected exception not thrown for query: ${queryString} ${JSON.stringify(queryArgs)}`,
+      );
+    });
+  };
+
+  /**
+   * Helper method which runs an array of queries and asserts the results' provided [propertyToCompare]
+   * field to equal the given expected result array's values.
+   * For example: (r, "Obj", "intCol" [[3, 4], "intCol > 2]) => querying "intCol > 2" should
+   * return results whose elements' intCol property values are 3 and 4.
+   */
+  const assertQueryResultValues = (
+    realm: Realm,
+    objectSchema: string | Realm.ObjectClass,
+    propertyToCompare: string,
+    queryResultPairs: QueryResultsPair[],
+  ) => {
+    queryResultPairs.forEach(([expectedResults, queryString, ...queryArgs]) => {
+      let results = realm.objects<any>(objectSchema);
+      results = results.filtered(queryString, ...queryArgs);
+      expect(expectedResults.length).equals(results.length);
+      expect(expectedResults).to.deep.equal(
+        results.map((el) => el[propertyToCompare]),
+        `
+        Expected results not returned from query: ${queryString} ${JSON.stringify(queryArgs)}, 
+      `,
+      );
+    });
+  };
+
+  beforeEach(Realm.clearTestState);
```

Comment:
When calling `openRealmBeforeEach` an after-each hook calling `clearTestState` will also be registered, in which case I'd expect this is not needed.
```suggestion
```